doc/MiniLanguageTutorial.md

```diff
@@ -480,37 +563,278 @@ Mini never automatically converts types to make an operation succeed.  Programme
+* `fixed array`:  
+  A default value for a `fixed array` of type *type* and length *length* is a fixed array of length *length*, where every array element has a value of the default value of *type*.
+  See the AVM representation section for more information of how the memory for this value is layed out.
+* `nominal`, `specialized generic`:  
+  The default value for a nominal or specialized generic type is the default value of its representation.
+
+While non-specialized generic types should never be created, if they are they will have a default value of `()`.
+
+## AVM Representation
+
+Types in mini are internally represented as AVM values.
+
+1. uint:  
+   Always an `Int`, any `Int` value is allowable for this type.
+1. int:  
+   Always an `Int`, any `Int` value is allowable for this type.
+1. bool:  
+   Always an `Int`, the values `0` and `1` are allowed for this type, but nothing else.
+1. buffer:  
+   Always a `Buffer`, any value is allowable for this type.
+1. bytes32:  
+   Always an `Int`, any value is allowable for this type.
+1. address:  
+   Always an `Int`, only values up to `2^160-1` are allowed for this type, where the `Int` must be interpreted as unsigned.
+1. void:  
+   Represents lack of a value, whenever a value of type `void` is present, then it is assumed there is no value there.
+1. struct:  
+   Represented by a wrapping tuple containing values of the types of each of its fields, in order from top to bottom.
+   Field names have no impact on the allowable values for the type.
+1. tuple:  
+   Identical to structs, represented by a wrapping tuple containing values of the types of each of its fields, in order left to right.
+1. unsized array of *type*:  
+   The depth of an unsized array is the smallest integer `x` such that 8^`x` > *length*, with a minimum depth of `1`, where *length* is the current length of the array.
+   An array of depth `N` has a value of `Tuple(Int, Int,` a `Tuple` of length 8 of sized arrays of depth *N-1* `)`, where depth `0` represents an AVM value valid for *type*.
+   The first `Int` in the outer tuple must be the *length* of the array, and the second integer must be `8^(N-1)`.
+1. sized array of *type* and length *length*:  
+   A series of nested 8 tuples, each leaf of which contains a AVM value valid for *type*.
+   The nested tuples are at uniform depth, and are the minimum depth such that there are enough slots for *length* values.
+   The minimum depth is a single 8 tuple. Sized arrays of length 0 allocate a single tuple with 8 slots.
+1. map with keys of *keytype* and values of *valuetype*:  
+   All maps consist of `Tuple(` *KvsNode* `, Int)`, the valid values for *KvsNode* will be explained below.
+   A *KvsNode* may be one of 3 types, it may be an integer of value `0`, a `Tuple` of length 2, or a `Tuple` of length 8.
+   In the case of a 2 `Tuple`, the first value in the tuple must be valid for *keytype*, and the second must be valid for `option<`*valuetype*`>`.
```

Comment:
Remember, this section is not just saying which representations are valid and which are not, it is also describing how to determine the representation for any given value.  So it's not enough to say that it can be a 2-tuple with certain types in the field, you also need to say what map that corresponds to.

Reply:
While I think this is useful information, I feel like I should mention that I did not state that this was part of the scope for the section.

Reply:
I have added this information.